### ISSUE TITLE
Possible localization bug in BasicBlockListModel?

### ISSUE THREAD

**Issue by rogalmic** [nikcio/Nikcio.UHeadless#91]
https://github.com/nikcio/Nikcio.UHeadless/blob/ada63c4cde62b5bc32137e61ebd680c75f11e642/src/Nikcio.UHeadless.Base/Basics/EditorsValues/BlockList/Models/BasicBlockListModel.cs#L34

Should this line pass culture like so?
```
var propertyValue = createPropertyValue.Property.GetValue(createPropertyValue.Culture);
```

I was getting cryptic error when querying blocklist, seems it was related to the culture being null:
![image](https://user-images.githubusercontent.com/23117003/193864710-8e17facf-3447-4904-8990-d52523208e72.png)
```
   at Umbraco.Cms.Core.Collections.CompositeStringStringKey..ctor(String key1, String key2)
   at Umbraco.Cms.Infrastructure.PublishedCache.Property.CacheValues.For(String culture, String segment)
   at Umbraco.Cms.Infrastructure.PublishedCache.Property.GetValue(String culture, String segment)
```

**Comment by nikcio** [comment]
Hi @rogalmic 

Could you add what version of UHeadless and Umbraco CMS you experience this error on. And any reproduction steps that might be helpful debugging the issue.

**Comment by rogalmic** [comment]
Hi @nikcio 

UHeadless version:
```
<PackageReference Include="Nikcio.UHeadless" Version="3.0.2" />
```

It is hard for me to create a test app reproducing this, since i added this as a test to already existing 2 language umbraco app.

The graphql will return following error for "Umbraco.BlockList" elements:
```
      "extensions": {
        "message": "Exception has been thrown by the target of an invocation.",
        "stackTrace": "   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)\r\n   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)\r\n   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)\r\n   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)\r\n   at Nikcio.UHeadless.Core.Reflection.Factories.DependencyReflectorFactory.GetReflectedType[T](Type typeToReflect, Object[] constructorRequiredParamerters)\r\n   at lambda_method142(Closure , IPureResolverContext )\r\n   at HotChocolate.Execution.Processing.Tasks.ResolverTaskFactory.<ResolveAndCompleteInline>g__TryExecute|4_0(Object& result, <>c__DisplayClass4_0& )"
      }
```

To find the actual root cause, i created a custom PropertyMap  @ PropertyMapOptions, which allowed to replace default handler with my own copy implementation of  BasicBlockListModel. I found out that `createPropertyValue.Culture` has proper culture string, but it is not used in GetValue call "key1 is null", which results in exception from original report.


Side questions:
- is there a way to trace also inner exceptions, since the graphql returned one does not provide useful info?
- maybe it is possible to replace already existing entry in PropertyMap @ PropertyMapOptions, since AddEditorMapping will only add handler if umbraco type handler is not there (hence i needed to add custom AddEditorMapping implementation to register my "Umbraco.BlockList" handler in place of default one):
```
    public class PropertyMapExt : PropertyMap {
        public override void AddEditorMapping<TType>(string editorName)
        {
            base.types.Remove(typeof(TType));
            base.editorPropertyMap.Remove(editorName);
            base.AddEditorMapping<TType>(editorName);
            base.AddUsedType<TType>();
        }
    }
```

**Comment by nikcio** [comment]
Hi @rogalmic,

It looks like a mistake that `createPropertyValue.Culture` wasn't included in the `getValue` call. I've included it now in v3.0.3/3.0.4 and in my testing, it doesn't make a difference for the query. But with a way to reproduce the error, I cannot be certain that the error is fixed by this change. Please try updating to the new version and see if the error is resolved.

### Side questions:

#### Is there a way to trace also inner exceptions, since the graphql returned one does not provide useful info?

Right now there's no better way to trace the errors than in the GraphQL returned one. You're welcome to create a PR implementing some kind of logging or other way to return the errors to the developers.

A starting point might be here: https://github.com/nikcio/Nikcio.UHeadless/blob/fdef65eff44993a6e194f8e1497ce8298b86b3f8/src/Nikcio.UHeadless.Core/Reflection/Factories/DependencyReflectorFactory.cs#L41
Here most errors on properties will go through I don't know if there's a great way to handle it here.


#### Maybe it is possible to replace already existing entry in PropertyMap @ PropertyMapOptions, since AddEditorMapping will only add handler if umbraco type handler is not there (hence i needed to add custom AddEditorMapping implementation to register my "Umbraco.BlockList" handler in place of default one):

I've fixed this in v3.0.4. Now custom mappings will overrule.
